src/common/commonutils/SshUtils.c

```diff
@@ -220,171 +235,180 @@ int CheckAppropriateCiphersForSsh(const char** ciphers, unsigned int numberOfCip
-        return status;
+        OsConfigLogError(log, "CheckSshClientAliveInterval: 'clientaliveinterval' is not set to a greater than zero value in SSH Server response (but to %d)", actualValue);
+        OsConfigCaptureReason(reason, "'clientaliveinterval' is not set to a greater than zero value in SSH Server response (but to %d)",
+            "%s, also 'clientaliveinterval' is not set to a greater than zero value in SSH Server response (but to %d)", actualValue);
+        status = ENOENT;
     }
 
-    if (NULL != (value = GetSshServerState(option, log)))
-    {
-        OsConfigLogInfo(log, "CheckSshOptionIsSetToString: '%s' found in SSH Server response set to '%s'", option, value);
+    OsConfigLogInfo(log, "CheckSshClientAliveInterval: %s (%d)", status ? "failed" : "passed", status);
 
-        if (0 != strcmp(value, expectedValue))
-        {
-            OsConfigLogError(log, "CheckSshOptionIsSetToString: '%s' is not set to '%s' in SSH Server response (but to '%s')", option, expectedValue, value);
-            OsConfigCaptureReason(reason, "'%s' is not set to '%s' in SSH Server response (but to '%s')",
-                "%s, also '%s' is not set to '%s' in SSH Server response (but to '%s')", option, expectedValue, value);
-            status = ENOENT;
-        }
+    return status;
+}
 
-        FREE_MEMORY(value);
-    }
-    else
+int CheckSshLoginGraceTime(char** reason, void* log)
+{
+    int actualValue = 0;
+    int status = CheckSshOptionIsSetToInteger("logingracetime", NULL, &actualValue, reason, log);
+
+    if ((0 == IsSshServerActive(log)) && (actualValue > 60))
     {
-        OsConfigLogError(log, "CheckSshOptionIsSetToString: '%s' not found in SSH Server response", option);
-        OsConfigCaptureReason(reason, "'%s' not found in SSH Server response", "%s, also '%s' is not found in SSH server response", option);
+        OsConfigLogError(log, "CheckSshLoginGraceTime: 'logingracetime' is not set to 60 or less in SSH Server response (but to %d)", actualValue);
+        OsConfigCaptureReason(reason, "'logingracetime' is not set to a value of 60 or less in SSH Server response (but to %d)",
+            "%s, also 'logingracetime' is not set to a value of 60 or less in SSH Server response (but to %d)", actualValue);
         status = ENOENT;
     }
 
-    OsConfigLogInfo(log, "CheckSshOptionIsSetToString: %s (%d)", status ? "failed" : "passed", status);
+    OsConfigLogInfo(log, "CheckSshLoginGraceTime: %s (%d)", status ? "failed" : "passed", status);
 
     return status;
 }
 
-int CheckSshOptionIsSetToInteger(const char* option, int expectedValue, int* actualValue, char** reason, void* log)
+int SetSshOption(const char* option, const char* value, void* log)
 {
-    char* value = NULL;
-    int integerValue = 0;
+    // Replaces any instances of 'option foo' with 'option value' and adds 'option value' when 'option' is missing
+    const char* commandTemplate = "sed '/^%s /{h;s/ .*/ %s/};${x;/^$/{s//%s %s/;H};x}' %s";
```

Comment:
is `sed` available in all our supported distros?

Reply:
Yes. It's using on purpose not complications such as -i.